src/components/RepoSearchForm.tsx

```diff
@@ -11,33 +11,53 @@ import Search from 'antd/lib/input/Search';
 export const GithubSearchRepositories: React.FC = () => {
     const dispatch: any = useDispatch();
     const [searchName, setSearchName] = useState('');
-    const { list, isLoading } = useSelector((state: StoreState) => state.gitHubRepositories);
-
+    const { list, isLoading, total_repositories_count } = useSelector((state: StoreState) => state.gitHubRepositories);
     const onChangeRepositoryTitle = (e: ChangeEvent<HTMLInputElement>) => setSearchName(e.target.value);
+    const [pageSize, setPageSize] = useState<number>(10);
 
     const onClick = () => {
-        dispatch(searchGithubRepositories(searchName));
+        dispatch(searchGithubRepositories(searchName, 1, pageSize));
+    };
+    const handlePagination = (currentPage: number) => {
+        dispatch(searchGithubRepositories(searchName, currentPage, pageSize));
+    };
+    const handleShowSizeChange = (currentPage: number, pageSize: number) => {
+        setPageSize(pageSize);
+        dispatch(searchGithubRepositories(searchName, currentPage, pageSize));
     };
 
     return (
-            <div className="mainContainer">
-                <Form>
-                    <h2 className="formHeaderTitle">Search Repositories</h2>
-                    <Search
-                            className="searchInput"
-                            placeholder="Enter repository name"
-                            enterButton="Search"
-                            size="large"
-                            value={searchName}
-                            loading={isLoading}
-                            onSearch={onClick}
-                            onChange={onChangeRepositoryTitle}
-                    />
-                </Form>
+        <div className="mainContainer">
+            <Form>
+                <h2 className="formHeaderTitle">Search Repositories</h2>
+                <Search
+                    className="searchInput"
+                    placeholder="Enter repository name"
+                    enterButton="Search"
+                    size="large"
+                    value={searchName}
+                    loading={isLoading}
+                    onSearch={onClick}
+                    onChange={onChangeRepositoryTitle}
+                />
+            </Form>
+            <Spin className="loadingSpin" spinning={isLoading} size="large" tip="Loading...">
                 <div className="itemsContainer">
                     {list.map((item: GithubRepositoryItem) => <RepositoriesItems key={item.id} item={item} />)}
                 </div>
-                {/*<Pagination defaultCurrent={1} total={50} />*/}
-            </div>
+            </Spin>
+            {
+                list.length ?
+                    <Pagination
+                        className="paginationBar"
+                        showSizeChanger
+                        onShowSizeChange={handleShowSizeChange}
```

Comment:
Keep event handler naming consistent across application: it's `handleSomething` either `onSomething`.